File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXFlatButton.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.scene.paint.Color;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import org.develnext.jphp.ext.javafx.support.control.FlatButton;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Nullable;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NS + "UXFlatButton")
public class UXFlatButton extends UXButtonBase<FlatButton> {
    interface WrappedInterface {
        @Property Color color();
        @Property @Nullable Color hoverColor();
        @Property @Nullable Color clickColor();
        @Property double borderRadius();
    }

    public UXFlatButton(Environment env, FlatButton wrappedObject) {
        super(env, wrappedObject);
    }

    public UXFlatButton(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Reflection.Signature
    public void __construct() {
        __wrappedObject = new FlatButton();
    }

    @Reflection.Signature
    public void __construct(String text) {
        __wrappedObject = new FlatButton(text);
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/UXFlatButton.php
<?php
namespace php\gui;

use php\gui\paint\UXColor;

/**
 * Class UXFlatButton
 * @package php\gui
 *
 * @packages gui, javafx
 */
class UXFlatButton extends UXButtonBase
{
    /**
     * @var UXColor
     */
    public $color;

    /**
     * @var UXColor
     */
    public $hoverColor;

    /**
     * @var UXColor
     */
    public $clickColor;

    /**
     * @var float
     */
    public $borderRadius;
}